server/src/main/java/org/apache/iotdb/db/query/control/FileReaderManager.java

```diff
@@ -199,6 +202,26 @@ void decreaseFileReaderReference(TsFileResource tsFile, boolean isClosed) {
    * integration tests will not conflict with each other.
    */
   public synchronized void closeAndRemoveAllOpenedReaders() throws IOException {
+    Iterator<Map.Entry<TsFileResource, TsFileSequenceReader>> iterator = closedFileReaderMap.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry<TsFileResource, TsFileSequenceReader> entry = iterator.next();
+      entry.getValue().close();
+      if (resourceLogger.isInfoEnabled()) {
+        resourceLogger.info("{} closedTsFileReader is closed.", entry.getValue().getFileName());
+      }
+      closedReferenceMap.remove(entry.getKey());
+      iterator.remove();
```

Comment:
duplicated?

Reply:
different map. iterator is the closedFileReaderMap's iterator.